lightning/src/onion_message/packet.rs

```diff
@@ -114,7 +114,7 @@ pub(super) enum Payload<T: CustomOnionMessageContents> {
 	}
 }
 
-#[derive(Debug)]
+// #[derive(Debug)]
```

Comment:
Keep the `Debug` derive and remove the custom implementation as per earlier discussion.

Reply:
I have been busy with my End term examination of my college , so i wasn't able to update this PR.
So if i use Derive Debuginstead of custom implementation, how to implement specific formatting logic 
`match self {
            ParsedOnionMessageContents::Offers(ref msg) => {
                write!(f, " (Offers): {:?}", msg)
            }
            ParsedOnionMessageContents::Custom(ref msg) => {
                write!(f, " (Custom) : {}", msg.tlv_type())
            }
        }`
Thanks:)

Reply:
Not sure if I understand your question. You don't need the custom formatting logic so long as you also addressed https://github.com/lightningdevkit/rust-lightning/pull/2642#discussion_r1380177171.